apps/fetch.py

```diff
@@ -15,12 +15,26 @@
 from helpers import cmd
 
 
-def fetch_target_apps(targets: dict, apps_shortlist: str, token: str, dst_dir: str):
+def fetch_target_apps(targets: dict, apps_shortlist: set[str], token: str, dst_dir: str) -> dict[str, set[str]]:
     apps_fetcher = SkopeAppFetcher(token, dst_dir)
+    fetched_target_apps: dict[str, set[str]] = {}
     for target_name, target_json in targets.items():
-        apps_fetcher.fetch_target(FactoryClient.Target(target_name, target_json),
-                                  [x.strip() for x in apps_shortlist.split(',') if x]
-                                  if apps_shortlist else None, force=True)
+        target = FactoryClient.Target(target_name, target_json)
+        target_apps = set(app for app, _ in target.apps())
+        target_app_shortlist: set[str] = None
+        if apps_shortlist:
+            target_app_shortlist = target_apps.intersection(apps_shortlist)
```

Comment:
should we log what got dropped from this intersection so that its easier to detect if someone made a typo?

Reply:
Yep, I think it may be useful. It's not necessary has to be type. Customers may have different set of target apps in different branches. I will add the log.

Reply:
Updated.